### DISCUSSION TITLE
[help] Running bespoke functions on error before stopping

### DISCUSSION THREAD

**Discussion by rsangole** [ropensci/targets#1261] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Is it possible to define a bespoke function that get's executed on an error within a targets pipeline, for example:  "for this target, if an error occurs, run `update_slack_thread_with_error(err)` before stopping the targets pipeline" ?

Similar to what I'd do in the `error` and `interrupt` arguments when using a `tryCatch()` framework:

```r
  tryCatch(
        expr = {
                # do stuff here...
                # ...
        },
        error = function(err) {
            update_slack_thread_with_error(err)
        },
        interrupt = function(intr) {
            update_slack_thread_with_error(intr)
        }
  )
```

**Comment by wlandau** [comment]
I have no plans to implement this directly in `targets`, but you can run a `tryCatch()` block in the command of your target or the custom function it calls.
